### DISCUSSION TITLE
Using draw.io to draw the SVG

### DISCUSSION THREAD

**Discussion by torwag** [ExperienceLovelace/ha-floorplan#443] [General]
Hi,
I used draw.io for other projects related to SVG graphics, and I was wondering if someone here tried it for ha-floorplan. 
Is there a pro or con for using draw.io?
Just curious, before I start to figure out what everyone else knows already.

**Comment by OzGav** [comment]
Sorry don’t recall anyone else mentioning it. Try a search? I use Inkscape with a PNG created by SweetHome 3D

**Reply by exetico** [reply]
I've not tried that myself, either. @torwag : Are you able to change the id and reference of each SVG element? I guess that's the most important thing. Also, I guess most topics about "how to do X and Y", would only give you inputs on how to solve it in Inkscape. But, give it a go. I mean, you can try and replicate something as simple as this, just to see how it goes:

 https://youtu.be/f_3KAPc53hQ?si=T4YYfdT5oV66EEZy&t=145

**Comment by torwag** [comment]
As feedback, I gave draw.io a try. The program itself is rather easy to use, and I could see how people could create nice process flow diagrams. Unfortunately, it seems that the SVG-export is rather non-standard. There is a lot of additional information in the SVG and even cleaning it up didn't help. Furthermore, it seems albeit you can name the objects IDs in draw.io, they are not exported in the SVG file. There is a plugin svgdata which promise to export the IDs, but it appends each ID with "cell-", which makes it more complex, and it didn't work either. The elements in the exported SVG-files seem not to be of standard or regular type. E.g. in inkscape you get a text-element for a text entry. In the exported SVG of draw.io a text will be embedded in some  \<foreignObject\> elements.

Thus, it seems it will not work for ha-floorplan.
